### PR TITLE
📊 : Import displacement data from IDMC

### PR DESCRIPTION
**Pull request by antea04**
Part III of owid/owid-issues#606

Includes:

From the [IDMC:](https://www.internal-displacement.org/database/displacement-data/)
- [x] Import of internal displacements due to conflict & disaster
- [x] Import of displaced people stock due to conflict & disaster
- [x] Share of people displaced due to conflict & disaster
- [x] Internal displacements per 100 people (due to conflict & disaster)

### CODE REVIEW

File: etl/steps/data/meadow/idmc/2024-08-02/internal_displacement.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        df=tb, countries_file=paths.country_mapping_path, warn_on_missing_countries=True, warn_on_unused_countries=True
+    )
+
+    # calculate population averages
+    tb = geo.add_population_to_table(tb, ds_pop)
+
+    tb["total_stock_displacement"] = tb["conflict_stock_displacement"] + tb["disaster_stock_displacement"]
+    tb["total_internal_displacements"] = tb["conflict_internal_displacements"] + tb["disaster_internal_displacements"]
+
+    tb["share_of_internally_displaced_pop"] = tb["total_stock_displacement"] / tb["population"] * 100
```
Comment (veronikasamborska1994) [inline etl/steps/data/meadow/idmc/2024-08-02/internal_displacement.py:53]: nit - you could do this in a loop -e.g.,  
```

    columns_to_calculate = [
        ("share_of_internally_displaced_pop", "total_stock_displacement"),
        ("share_of_conflict_displaced_pop", "conflict_stock_displacement"),
        ("share_of_disaster_displaced_pop", "disaster_stock_displacement"),
        ("displacements_per_100_people", "total_internal_displacements"),
        ("conflict_displacements_per_100_people", "conflict_internal_displacements"),
        ("disaster_displacements_per_100_people", "disaster_internal_displacements"),
    ]

    for new_column, source_column in columns_to_calculate:
        tb[new_column] = (tb[source_column] / tb["population"]) * 100
```

File: etl/steps/data/meadow/idmc/2024-08-02/internal_displacement.py
Comment (veronikasamborska1994) [inline etl/steps/data/meadow/idmc/2024-08-02/internal_displacement.py:1]: usually we do these sort of calculations in the garden step but I personally don't think it makes much of a difference so up to you if you want to move it or not!

File: etl/steps/data/meadow/idmc/2024-08-02/internal_displacement.meta.yml [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+          [Internally displaced people](#dod:internally-displaced) due to both conflict and violence and natural disaster. This includes all people who are currently displaced, no matter which year they originally got displaced.
+        unit: people
+        short_unit: ""
+        description_from_producer: |-
+          - {definitions.idmc_definitions.internally_displaced_people}
+          - {definitions.idmc_definitions.conflict}
+          - {definitions.idmc_definitions.disaster}
+        description_processing: This figure is the sum of the number of people displaced due to conflict and violence and the number of people displaced due to natural disasters.
+      share_of_internally_displaced_pop:
+        title: Share of internally displaced people of total population
```
Comment (veronikasamborska1994) [inline etl/steps/data/meadow/idmc/2024-08-02/internal_displacement.meta.yml:44]: nit - this title sounds a bit weird to me and I'd maybe rephrase it to "Internally displaced people as a share of total population"

File: etl/steps/data/meadow/idmc/2024-08-02/internal_displacement.meta.yml [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+        title: Internally displaced people due to conflict
+        description_short: |-
+          [Internally displaced people](#dod:internally-displaced) due to conflict. This includes all people who are currently displaced due to conflict, no matter which year they originally got displaced.
+        unit: people
+        short_unit: ""
+        description_from_producer: |-
+          - {definitions.idmc_definitions.internally_displaced_people}
+          - {definitions.idmc_definitions.conflict}
+      share_of_conflict_displaced_pop:
+        title: Share of internally displaced people due to conflict of total population
```
Comment (veronikasamborska1994) [inline etl/steps/data/meadow/idmc/2024-08-02/internal_displacement.meta.yml:67]: same here - maybe "Internally displaced people due to conflict as a share of total population" but depends if you want to use these as the actual titles for charts. If not then you don't really need to worry about it and the titles of the variables make sense

File: etl/steps/data/meadow/idmc/2024-08-02/internal_displacement.meta.yml [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+        title: Internally displaced people due to disaster
+        description_short: |-
+          [Internally displaced people](#dod:internally-displaced) due to natural disasters. This includes all people who are currently displaced due to disasters, no matter which year they originally got displaced.
+        unit: people
+        short_unit: ""
+        description_from_producer: |-
+          - {definitions.idmc_definitions.internally_displaced_people}
+          - {definitions.idmc_definitions.disaster}
+      share_of_disaster_displaced_pop:
+        title: Share of internally displaced people due to disaster of total population
```
Comment (veronikasamborska1994) [inline etl/steps/data/meadow/idmc/2024-08-02/internal_displacement.meta.yml:88]: same here but again up to you if you want to change it here or later in the charts

**Review by veronikasamborska1994 — APPROVED** [review]
LGTM!

File: etl/steps/data/meadow/idmc/2024-08-02/internal_displacement.meta.yml (hunk 2 above)
Comment (antea04) [reply etl/steps/data/meadow/idmc/2024-08-02/internal_displacement.meta.yml:44]: That's a good idea - thank you!

File: etl/steps/data/meadow/idmc/2024-08-02/internal_displacement.py (hunk 1 above)
Comment (antea04) [reply etl/steps/data/meadow/idmc/2024-08-02/internal_displacement.py:53]: Nice, thank you! I switched it to a loop now :D
